reasoner/processor/AbstractProcessor.java

```diff
@@ -156,27 +153,7 @@ public void onFinished(Identifier finishable) {
 
     @Override
     public void exception(Throwable e) {
-        if (e instanceof TypeDBException && ((TypeDBException) e).code().isPresent()) {
-            String code = ((TypeDBException) e).code().get();
-            if (code.equals(RESOURCE_CLOSED.code())) {
-                LOG.debug("Processor interrupted by resource close: {}", e.getMessage());
-                controller.execute(actor -> actor.exception(e));
-                return;
-            } else {
-                LOG.debug("Processor interrupted by TypeDB exception: {}", e.getMessage());
-            }
-        }
-        LOG.error("Actor exception", e);
-        controller.execute(actor -> actor.exception(e));
-    }
-
-    public void terminate(Throwable cause) {
-        LOG.debug("Actor terminated.", cause);
-        this.terminated = true;
-    }
-
-    private boolean isTerminated() {
-        return terminated;
+        controller.executePreemptive(controller -> controller.exception(e));
```

Comment:
we let the `Controller` plane decide what to do with each exception - they will propagate termination commands to the registry (eg. other controllers) and their processors (including this one).